ccaas_builder/cmd/detect/main.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+var logger = log.New(os.Stderr, "", 0)
+
+func main() {
+	logger.Println("::Detect")
+
+	if err := run(); err != nil {
+		logger.Printf("::Error: %v\n", err)
+		os.Exit(1)
+	}
+
+	logger.Printf("::Type detected as ccaas")
+
+}
+
+type chaincodeMetadata struct {
+	Type string `json:"type"`
+}
+
+func run() error {
+	if len(os.Args) < 3 {
+		return fmt.Errorf("too few arguments")
+	}
+
+	chaincodeMetaData := os.Args[2]
+
+	metadataFile := filepath.Join(chaincodeMetaData, "metadata.json")
+	_, err := os.Stat(metadataFile)
+	if err != nil {
+		return fmt.Errorf("%s not found ", metadataFile)
```

Comment:
Even for these we usually use errors.New().

Reply:
just been updating these, but now have VSCode warnings...

```should use fmt.Errorf(...) instead of errors.New(fmt.Sprintf(...)) (S1028) go-staticcheck```